identity/src/commonMain/kotlin/com/android/identity/nfc/ServiceSelectRecord.kt

```diff
@@ -26,9 +29,9 @@ data class ServiceSelectRecord(
             }
 
             require(record.payload.size >= 1) { "Unexpected length of Service Select Record" }
-            val serviceNameLen = record.payload[0].toInt().and(0xff)
+            val serviceNameLen = record.payload.getUInt8(0).toInt()
             require(record.payload.size == serviceNameLen + 1) { "Unexpected length of body in Service Select Record" }
-            return ServiceSelectRecord(record.payload.toByteArray().decodeToString(1))
+            return ServiceSelectRecord(record.payload.decodeToString().drop(1))
```

Comment:
someByteStr.toByteArray().decodeToString(1) is subtly different from someByteStr.decodeToString().drop(1) - the former drops the first _byte_, the latter drops the first _character_.

They're _probably_ the same but you now force the reader to have to reason about this and what this calls for is, well, pick up the string starting at byte offset 1. Please change it back to using decodeToString(1).

Reply:
...nice catch.